app/connectors/SecureMessageFrontendConnector.scala

```diff
@@ -19,17 +19,25 @@ package connectors
 import scala.concurrent.{ExecutionContext, Future}
 import com.google.inject.Inject
 import config.AppConfig
-import models.{ConversationPartial, InboxPartial, MessageFilterTag, ReplyResultPartial}
+import models.{AuthKey, ConversationPartial, InboxPartial, MessageFilterTag, ReplyResultPartial}
 import play.api.Logging
 import play.api.http.Status
+import services.AuditService
 import uk.gov.hmrc.http.HttpReads.Implicits.readRaw
 import uk.gov.hmrc.http.{HeaderCarrier, HttpClient, HttpResponse, UpstreamErrorResponse}
 
-class SecureMessageFrontendConnector @Inject()(httpClient: HttpClient, config: AppConfig)(implicit ec: ExecutionContext) extends Logging with Status {
+class SecureMessageFrontendConnector @Inject()(httpClient: HttpClient, config: AppConfig, auditService: AuditService)(implicit ec: ExecutionContext)
+    extends Logging with Status {
 
   def retrieveInboxPartial(eori: String, filter: MessageFilterTag)(implicit hc: HeaderCarrier): Future[InboxPartial] =
-    fetchPartial(config.microservice.services.secureMessaging.fetchInboxEndpoint, "the user's inbox", constructInboxEndpointQueryParams(eori, filter))
-      .map(response => InboxPartial(response.body))
+    fetchPartial(
+      config.microservice.services.secureMessaging.fetchInboxEndpoint,
+      "the user's inbox",
+      constructInboxEndpointQueryParams(AuthKey.enrolment, eori, filter)
+    ).map { response =>
+      auditService.auditSecureMessageInbox(AuthKey.enrolment, eori, filter, config.microservice.services.secureMessaging.fetchInboxEndpoint)
```

Comment:
Since this audit method is specific to the retrieval of the inbox you could avoid to pass `fetchInboxEndpoint` but instead use it directly while composing the audit details (IMO).

Reply:
I pass the path for the same reason as the other parameters, if they ever change in the connector they might not be updated in the AuditDetails class. I also think this is cleaner, no need for AuditService to have an otherwise unnecessary dependency on the AppConfig class (and that also makes testing of AuditService easier too).

Reply:
:+1: I like the idea to remove the dependency. I did not notice before.